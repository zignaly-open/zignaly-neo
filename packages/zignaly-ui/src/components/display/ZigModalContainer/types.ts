import React, { ReactElement } from "react";
import { SxProps } from '@mui/system';

export type ModalContainerProps = {
  children: React.ReactElement[] | React.ReactElement;
  title?: ReactElement | string;
  titleAlign?: "center" | "left";
  width?: number;
  padding?: number;
  onClickClose?: () => void;
  onGoBack?: () => void;
  titleStyles?: SxProps;
};
