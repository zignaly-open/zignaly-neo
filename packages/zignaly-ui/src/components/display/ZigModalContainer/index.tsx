import React, { forwardRef } from "react";
import { useMediaQuery, useTheme } from "@mui/material";
import { BackIconButton, Body, CloseIconButton, Header, Inline, Layout, Title } from "./styles";
import { ModalContainerProps } from "./types";
import { ZigBackIcon, ZigCrossIcon } from "../../../icons";

const ModalContainer = forwardRef((props: ModalContainerProps, ref) => {
  const {
    children,
    title = null,
    titleAlign = "center",
    onGoBack = null,
    width,
    onClickClose = null,
    titleStyles,
    mobileFullScreen,
  } = props;
  const theme = useTheme();
  const sm = useMediaQuery(theme.breakpoints.up("sm"));
  const buttonSize = !sm && mobileFullScreen ? "22px" : "32px";

  return (
    <Layout id={"modal-container"} width={width as number} mobileFullScreen={mobileFullScreen} ref={ref} tabIndex={-1}>
      {onGoBack && typeof onGoBack === "function" && (
        <BackIconButton onClick={onGoBack}>
          <ZigBackIcon
            width={buttonSize}
            height={buttonSize}
            color={theme.palette.neutral100}
            id={"modal__back"}
          />
        </BackIconButton>
      )}
      <Header compact={!title && !onGoBack}>
        <Inline align={titleAlign}>
          {!!title && (
            <Title
              variant={mobileFullScreen && !sm ? "h2" : "h1"}
              mb={0}
              mt={mobileFullScreen && !sm ? "h2" : "h1" ? -2 : undefined}
              color="neutral100"
              id={"modal__title"}
              sx={titleStyles}
            >
              {title}
            </Title>
          )}
        </Inline>
      </Header>
      {onClickClose && typeof onClickClose === "function" && (
        <CloseIconButton onClick={onClickClose}>
          <ZigCrossIcon
            width={buttonSize}
            height={buttonSize}
            color={theme.palette.neutral100}
            id={"modal__close"}
          />
        </CloseIconButton>
      )}
      <Body mobileFullScreen={mobileFullScreen}>{children}</Body>
    </Layout>
  );
});

export { ZigModalActions, ZigModalForm } from "./styles";
export default ModalContainer;
