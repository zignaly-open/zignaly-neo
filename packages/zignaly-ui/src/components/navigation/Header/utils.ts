// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
