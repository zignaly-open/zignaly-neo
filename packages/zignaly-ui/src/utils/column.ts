import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
