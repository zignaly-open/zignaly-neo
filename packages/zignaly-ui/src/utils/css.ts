import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
