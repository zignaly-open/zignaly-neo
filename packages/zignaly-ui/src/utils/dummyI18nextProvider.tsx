import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
