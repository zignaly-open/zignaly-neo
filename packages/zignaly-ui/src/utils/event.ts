export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
