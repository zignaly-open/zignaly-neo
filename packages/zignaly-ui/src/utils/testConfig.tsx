import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
