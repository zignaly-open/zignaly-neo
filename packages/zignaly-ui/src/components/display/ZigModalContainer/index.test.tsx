import React from "react";
import ZigModalContainer from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { ZigTypography } from "../../../index";
import { fireEvent, waitFor } from "@testing-library/react";

describe("components/display/ZigModalContainer", () => {
  describe("renders", () => {
    it("ZigModalContainer render", async () => {
      const { container } = renderWithProvidersUi(
        <ZigModalContainer>
          <ZigTypography>modal</ZigTypography>
        </ZigModalContainer>,
      );

      const modal = container.querySelector("#modal-container") as Element;
      expect(modal).toBeVisible();
      const modalStyles = getComputedStyle(modal);
      expect(modal).toMatchSnapshot();
      expect(modalStyles).toMatchSnapshot();
    });
  });
  describe("actions", () => {
    it("ZigModalContainer close", async () => {
      const onClick = jest.fn();
      const { container } = renderWithProvidersUi(
        <ZigModalContainer onClickClose={onClick}>
          <ZigTypography>modal</ZigTypography>
        </ZigModalContainer>,
      );

      const closeButton = container.querySelector("#modal__close") as Element;
      const goBackButton = container.querySelector("#modal__back") as Element;
      expect(goBackButton).not.toBeInTheDocument();
      fireEvent.click(closeButton);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });
    it("ZigModalContainer go back", async () => {
      const onClick = jest.fn();
      const { container } = renderWithProvidersUi(
        <ZigModalContainer onGoBack={onClick}>
          <ZigTypography>modal</ZigTypography>
        </ZigModalContainer>,
      );

      const goBackButton = container.querySelector("#modal__back") as Element;
      fireEvent.click(goBackButton);
      await waitFor(() => {
        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });
  });
});
