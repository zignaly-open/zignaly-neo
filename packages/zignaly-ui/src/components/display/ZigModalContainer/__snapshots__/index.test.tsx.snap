// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigModalContainer should render without crashing 1`] = `
<div
  class="MuiBox-root css-1na0hc3"
  id="modal-container"
  tabindex="-1"
>
  <div
    class="MuiBox-root css-8tssri"
  >
    <div
      class="css-1ethbpr"
    />
  </div>
  <div
    class="css-99csgg"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 css-15f8135-MuiTypography-root"
    >
      modal
    </span>
  </div>
</div>
`;

exports[`components/display/ZigModalContainer should render without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "position",
  "10": "visibility",
  "2": "flex-direction",
  "3": "justify-content",
  "4": "align-items",
  "5": "background",
  "6": "border",
  "7": "border-radius",
  "8": "padding",
  "9": "max-width",
  "_importants": {
    "align-items": "",
    "background": "",
    "border": "",
    "border-radius": "",
    "display": "",
    "flex-direction": "",
    "justify-content": "",
    "max-width": "",
    "padding": "",
    "position": "",
    "visibility": undefined,
  },
  "_length": 11,
  "_onChange": [Function],
  "_values": {
    "align-items": "flex-start",
    "background": "rgb(16, 18, 37)",
    "background-color": "rgb(16, 18, 37)",
    "border": "1px solid #35334a",
    "border-bottom": "1px solid #35334a",
    "border-bottom-color": "#35334a",
    "border-bottom-style": "solid",
    "border-bottom-width": "1px",
    "border-color": "#35334a",
    "border-left": "1px solid #35334a",
    "border-left-color": "#35334a",
    "border-left-style": "solid",
    "border-left-width": "1px",
    "border-radius": "16px",
    "border-right": "1px solid #35334a",
    "border-right-color": "#35334a",
    "border-right-style": "solid",
    "border-right-width": "1px",
    "border-style": "solid",
    "border-top": "1px solid #35334a",
    "border-top-color": "#35334a",
    "border-top-style": "solid",
    "border-top-width": "1px",
    "border-width": "1px",
    "display": "flex",
    "flex-direction": "column",
    "justify-content": "center",
    "max-width": "100%",
    "padding": "40px 56px",
    "padding-bottom": "40px",
    "padding-left": "56px",
    "padding-right": "56px",
    "padding-top": "40px",
    "position": "relative",
    "visibility": "visible",
  },
}
`;
